numba_dpex/core/datamodel/models.py

```diff
@@ -40,18 +40,72 @@ def __init__(self, dmm, fe_type):
             ),
             (
                 "parent",
-                types.CPointer(fe_type.dtype, addrspace=fe_type.addrspace),
+                types.CPointer(types.pyobject, addrspace=fe_type.addrspace),
```

Comment:
The `parent` attribute of the numba's `ArrayModel` stores a reference to the original Python object which was unboxed into a numba native array type. Under the hood the attribute gets stored as a `PyObject *` in the corresponding `arystruct_t` instance.

In Numba-dpex, we have all along been using a `fe_type.dtype` pointer to store the `parent` Python object pointer. It never caused any issue as the `parent` attribute is never used inside a kernel. However, the change here gets numba-dpex in sync with how numba's `ArrayModel` is defined and ensures that the data model definition here correctly advertises the attribute as a `PyObject *` rather than a `fe_type.dtype` pointer.